pkg/service/volume_service_test.go

```diff
@@ -101,6 +101,52 @@ func (suite *VolumeServiceTestSuite) TestGetNonExistentVolumeByID() {
 	}
 }
 
+func (suite *VolumeServiceTestSuite) TestGetVolumesPagination() {
+	arg := new(param.GetParams)
+	//arg.Page = new(pagination.Page)
+	pagination := new(pagination.Page)
+	// set batch size
+	pagination.SetPageSize(2)
+
+	arg.Page = pagination
+	// fetch only 2 volumes
+	volumes, err := suite.volumeService.GetVolumes(arg)
+	if err != nil {
+		suite.T().Errorf("TestGetVolumesPagination(): Unable to fetch volumes, err: %v", err.Error())
+		return
+	}
+
+	if len(volumes) != 2 {
+		suite.T().Errorf("TestGetVolumesPagination(): pagination count doesnt match with # of return volumes ")
+		return
+	}
+
+	// Batch processing
+	for arg.Page.NextPage() {
+		volumes, err := suite.volumeService.GetVolumes(arg)
```

Comment:
We could avoid duplicate code for getting the first page vs. the remaining pages like this:

	for hasNextPage := true; hasNextPage; hasNextPage = arg.Page.NextPage() {
		_, err := suite.volumeService.GetVolumes(arg)
		if err != nil {
			suite.T().Errorf("TestGetVolumesPagination(): Unable to fetch volumes, err: %v", err.Error())
			return
		}

		if len(volumes) > 2 {
			suite.T().Errorf("TestGetVolumesPagination(): Returned volumes are more than the requested page size")
			return
		}
	}